lib/travis/build/script/csharp.rb

```diff
@@ -204,10 +204,10 @@ def mono_osx_url
         end
 
         def dotnet_osx_url
-          if config[:dotnet].include? "-preview"
-            return "https://dotnetcli.azureedge.net/dotnet/preview/Installers/#{config[:dotnet]}/dotnet-dev-osx-x64.#{config[:dotnet]}.pkg"
+          if config[:dotnet].include? "-preview" && !is_dotnet_after_2_0_prev_2?
+            return "https://dotnetcli.azureedge.net/dotnet/preview/Installers/#{config[:dotnet]}/dotnet-#{is_dotnet_after_2_0_prev_2? ? "sdk" : "dev"}-osx-x64.#{config[:dotnet]}.pkg"
```

Comment:
It is cleaner to have a separate method that returns the package name based on the .NET preview version, rather than making that decision inline with a ternary operator.